### DISCUSSION TITLE
Gemfile.lock not included in PR?

### DISCUSSION THREAD

**Discussion by tcg** [renovatebot/renovate#12478] [Request Help]
I'm starting here with this question, because I don't feel like I have enough information to create an Issue or present a "real problem" -- so I'm trying to narrow down my options and understanding. 

Based on the [Renovate for Ruby docs](https://docs.renovatebot.com/ruby/) , I see this line: 

> A PR is created which updates the Gemfile and Gemfile.lock in a single commit

We are using self-hosted Renovate in Gitlab. Using the `renovate/renovate` image, aimed at multiple repos. 

NPM stuff has been going great for a long time! 

We recently turned on Ruby support for a repo.

### For the Ruby repo, we're getting PRs for `Gemfile`, but those PRs *do not* contain updates for the `Gemfile.lock`. Subsequently, CI builds fail, because of the mismatch. 

I've done a lot of searching around, but haven't found a similar issue or any explicit mention of this behavior. As quoted, the docs seem to indicate that it should just happen. 

**Am I overlooking something otherwise obvious?** 

Is there a particular reason it would or should not generate a `Gemfile.lock` in the same PR with the `Gemfile` update? 

I don't see anything in the logs about skipping or failing, re: `Gemfile`.

**Comment by rarkins** [comment]
The Gemfile.lock should be included in nearly every case where the package.json is updated (it's pretty rare that the lock file would have no changes). Run with `LOG_LEVEL=debug` and ask Renovate to rebase that PR and then capture the log when it happens. Search for `bundle lock` in a log message

**Reply by tcg** [reply]
Did not expect such a quick response, and I'm incredibly grateful. =) 

I did not see this in other job runs, but upon rebase it makes the underlying issue clear -- though I'm not sure how to resolve. 

I can provide more if necessary, but I think this will suffice to show the relevant issue: 

```
INFO: Gemfile.lock update failed due to an unknown reason (repository=internal-tools/partner-portal-api, branch=renovate/loofah-2.x)
       "err": {
         "killed": false,
         "code": 1,
         "signal": null,
         "cmd": "bundle lock --update loofah",
         "stdout": "",
         "stderr": "/usr/local/ruby/3.0.2/lib/ruby/3.0.0/rubygems.rb:281:in `find_spec_for_exe': Could not find 'bundler' (1.17.3) required by your /tmp/renovate/repos/gitlab/internal-tools/partner-portal-api/Gemfile.lock. (Gem::GemNotFoundException)\nTo update to the latest version installed on your system, run `bundle update --bundler`.\nTo install the missing version, run `gem install bundler:1.17.3`\n\tfrom /usr/local/ruby/3.0.2/lib/ruby/3.0.0/rubygems.rb:300:in `activate_bin_path'\n\tfrom /usr/local/ruby/3.0.2/bin/bundle:23:in `<main>'\n",
         "message": "Command failed: bundle lock --update loofah\n/usr/local/ruby/3.0.2/lib/ruby/3.0.0/rubygems.rb:281:in `find_spec_for_exe': Could not find 'bundler' (1.17.3) required by your /tmp/renovate/repos/gitlab/internal-tools/partner-portal-api/Gemfile.lock. (Gem::GemNotFoundException)\nTo update to the latest version installed on your system, run `bundle update --bundler`.\nTo install the missing version, run `gem install bundler:1.17.3`\n\tfrom /usr/local/ruby/3.0.2/lib/ruby/3.0.0/rubygems.rb:300:in `activate_bin_path'\n\tfrom /usr/local/ruby/3.0.2/bin/bundle:23:in `<main>'\n",
         "stack": "Error: Command failed: bundle lock --update loofah\n/usr/local/ruby/3.0.2/lib/ruby/3.0.0/rubygems.rb:281:in `find_spec_for_exe': Could not find 'bundler' (1.17.3) required by your /tmp/renovate/repos/gitlab/internal-tools/partner-portal-api/Gemfile.lock. (Gem::GemNotFoundException)\nTo update to the latest version installed on your system, run `bundle update --bundler`.\nTo install the missing version, run `gem install bundler:1.17.3`\n\tfrom /usr/local/ruby/3.0.2/lib/ruby/3.0.0/rubygems.rb:300:in `activate_bin_path'\n\tfrom /usr/local/ruby/3.0.2/bin/bundle:23:in `<main>'\n\n    at ChildProcess.exithandler (child_process.js:383:12)\n    at ChildProcess.emit (events.js:400:28)\n    at ChildProcess.emit (domain.js:475:12)\n    at maybeClose (internal/child_process.js:1058:16)\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)"
```

My guess is that the version of Ruby/Bundler in the `renovate/renovate` image is not the same, and thus not compatible with our `Gemfile.lock`? 

Is it out of scope of this thread to ask: Is it as simple as changing our Bundler version? If so: To which? (I'm hoping the Ruby version itself is less relevant 🤞🏻)

**Reply by viceice** [reply]
you should probably switch to the renovate slim version with a docker, so renovate can run the required versions of ruby and bundler as docker sidecar.

We have a helm chart with can do this, we also have gitlab pipeline templates. Search the discussions for more samples.

**Reply by tcg** [reply]
Thanks so much! That seems to confirm that we need to change how we have Renovate set up. Dang. 

(I was doing some experimenting and indeed, it's not as simple as changing the bundler version in the lockfile. The Ruby version mismatch causes some trouble as well). 

I'll have a look around for setting up Self-hosted Gitlab with Ruby.   😬 

NPM setup was just too easy. I foolishly assumed other langs would be as striaght-forward to configure. Haha!

Thank you!

**Reply by viceice** [reply]
https://gitlab.com/renovate-bot/renovate-runner
Use this templates and your getting it done as easy as using npx.
